pkg/operator/quorumguardcontroller/quorumguardcontroller.go

```diff
@@ -88,14 +88,14 @@ func (c *QuorumGuardController) isInHAMode(ctx context.Context) (bool, error) {
 	}
 	// Added to make sure that iif ha mode is not set in infrastructure object
 	// we will get the default value configv1.FullHighAvailabilityMode
-	if infraData.Status.HighAvailabilityMode == "" {
-		klog.Infof("HA mode was not set in infrastructure resource setting it to default value %s", configv1.FullHighAvailabilityMode)
-		infraData.Status.HighAvailabilityMode = configv1.FullHighAvailabilityMode
+	if infraData.Status.ControlPlaneTopology == "" {
```

Comment:
I'm not sure if that's an option, they asked to remove that option from the enum